plugins/outputs/pulsar/pulsar.go

```diff
@@ -0,0 +1,378 @@
+package pulsar
+
+import (
+	"context"
+	"fmt"
+	"github.com/gofrs/uuid"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+var ValidTopicSuffixMethods = []string{
+	"",
+	"measurement",
+	"tags",
+}
+
+var zeroTime = time.Unix(0, 0)
+
+type (
+	Pulsar struct {
+		URL             string `toml:"url"`
+		Topic           string `toml:"topic"`
+		TopicTag        string `toml:"topic_tag"`
+		ExcludeTopicTag bool   `toml:"exclude_topic_tag"`
+		RoutingTag      string `toml:"routing_tag"`
+		RoutingKey      string `toml:"routing_key"`
+
+		//Client Options
+		AuthProvider               string `toml:"auth_provider"`
+		TLSAllowInsecureConnection bool   `toml:"insecure_skip_verify"`
+		TLSTrustCertsFilePath      string `toml:"tls_ca"`
+		TLSValidateHostname        bool   `toml:"tls_validate_host_name"`
+		TLSCertificatePath         string `toml:"tls_cert"`
+		TLSPrivateKeyPath          string `toml:"tls_key"`
+		AuthToken                  string `toml:"auth_token"`
+		ConnectionTimeout          string `toml:"connection_timeout"`
+		OperationTimeout           string `toml:"operation_timeout"`
+
+		//Producer Options
+		CompressionType         int    `toml:"compression_type"`
+		MaxPendingMessages      int    `toml:"max_pending_messages"`
+		HashingScheme           string `toml:"hashing_scheme"`
+		BatchingMaxPublishDelay string `toml:"batching_max_publish_delay"`
+		BatchingMaxMessages     uint   `toml:"batching_max_messages"`
+
+		tenantNameSpace string
+		producerCache   map[string]pulsar.Producer
+
+		Log telegraf.Logger `toml:"-"`
+
+		client pulsar.Client
+
+		serializer serializers.Serializer
+	}
+)
+
+var sampleConfig = `
+  [[outputs.pulsar]]
+ ## URLs of pulsar url
+  url = "pulsar://localhost:6650"
```

Comment:
Is this a sensible default? If so, please comment this line as follows
```suggestion
  # url = "pulsar://localhost:6650"
```
to indicate the default value of the parameter.